accessor/range.hpp

```diff
@@ -66,14 +89,16 @@ class range {
      * Creates a new range.
      *
      * @tparam AccessorParam  types of parameters forwarded to the accessor
-     *                        constructor
+     *                        constructor. It does not allow range itself.
```

Comment:
I would remove that comment and write:
```
@note  We use SFINAE to allow for a default copy and move constructor to be generated,
       so a `range` is trivially copyable if the `Accessor` is trivially copyable.
```
After that, I am fine with keeping this without checking for Visual Studio compiler.